src/universal/mutations/UpdateReflectionLocationMutation.js

```diff
@@ -0,0 +1,165 @@
+  retroPhaseItemId?: string
+};
+
+graphql`
+  fragment UpdateReflectionLocationMutation_team on UpdateReflectionLocationPayload {
+    reflection {
+      ...CompleteReflectionFrag @relay(mask: false)
+    }
+    
+    reflectionGroup {
+      id
+      meetingId
+      sortOrder
+      retroPhaseItemId
+      reflections {
+        ...CompleteReflectionFrag @relay(mask: false)  
+      }
+      title
+    }
+    oldReflectionGroup {
+      id
+      title
+    }
+  }
+`;
+
+const mutation = graphql`
+  mutation UpdateReflectionLocationMutation($reflectionGroupId: ID, $reflectionId: ID, $retroPhaseItemId: ID, $sortOrder: Float!) {
+    updateReflectionLocation(
+      reflectionGroupId: $reflectionGroupId, reflectionId: $reflectionId, retroPhaseItemId: $retroPhaseItemId, sortOrder: $sortOrder) {
+      ...UpdateReflectionLocationMutation_team @relay(mask: false)
+    }
+  }
+`;
+
+const handleRemoveReflectionFromGroup = (reflectionId, reflectionGroupId, store) => {
+  const reflectionGroup = store.get(reflectionGroupId);
+  if (!reflectionGroup) return;
+  safeRemoveNodeFromArray(reflectionId, reflectionGroup, 'reflections');
+};
+
+const moveGroupLocation = (reflectionGroupProxy, store) => {
+  const reflectionGroupId = reflectionGroupProxy.getValue('id');
+  const meetingId = reflectionGroupProxy.getValue('meetingId');
+  handleRemoveReflectionGroups(reflectionGroupId, meetingId, store);
+  handleAddReflectionGroups(reflectionGroupProxy, store);
+};
+
+const moveReflectionLocation = (reflection, reflectionGroup, oldReflectionGroupId, store) => {
+  moveGroupLocation(reflectionGroup, store);
+  if (!reflection) return;
+  const reflectionId = reflection.getValue('id');
+  handleRemoveReflectionFromGroup(reflectionId, oldReflectionGroupId, store);
+  handleAddReflectionToGroup(reflection, store);
+  handleRemoveEmptyReflectionGroup(oldReflectionGroupId, store);
+};
+
+export const updateReflectionLocationTeamUpdater = (payload, store) => {
+  const reflection = payload.getLinkedRecord('reflection');
+  const reflectionGroup = payload.getLinkedRecord('reflectionGroup');
+  const oldReflectionGroupId = getInProxy(payload, 'oldReflectionGroup', 'id');
+  moveReflectionLocation(reflection, reflectionGroup, oldReflectionGroupId, store);
+};
+
+type Context = {
+  meetingId: string
+};
+
+const UpdateReflectionLocationMutation = (
+  environment: Object,
+  variables: Variables,
+  context: Context,
+  onError?: ErrorHandler,
+  onCompleted?: CompletedHandler
+) => {
+  return commitMutation(environment, {
+    mutation,
+    variables,
+    onCompleted,
+    onError,
+    updater: (store) => {
+      const payload = store.getRootField('updateReflectionLocation');
+      if (!payload) return;
+      updateReflectionLocationTeamUpdater(payload, store);
+
+      // only do this for the mutator, don't want unexpected collapses because someone else did something
+      const reflectionGroupId = getInProxy(payload, 'reflectionGroup', 'id');
```

Comment:
here we're collapsing the group just for the person that did the drag because group expansion should not be shared across users